client/mysqldump.cc

```diff
@@ -2759,9 +2774,9 @@ static void skip_secondary_keys(char *create_str, bool has_pk) noexcept {
           my_strndup(PSI_NOT_INSTRUMENTED, ptr, end - ptr + 1, MYF(MY_FAE));
 
       if (type == key_type_t::CONSTRAINT)
-        alter_constraints_list->emplace_front(data);
+        alter_constraints_list.emplace_back(data);
```

Comment:
Why s/front/back?

Reply:
You removed 
```
    alter_constraints_list= list_reverse(alter_constraints_list);
```
in `dump_skipped_keys`.

Reply:
That's right, but haven't I also changed which end of the list gets inserted to at the same time?

Reply:
`list_cons()` in 5.7 adds elements to the front of a list so it should be replaced with `emplace_front` + list_reverse or `emplace_back` without reversing.